diff -rwU5 A/Libsystem-1252/Libsystem.xcodeproj/project.pbxproj B/Libsystem-1252/Libsystem.xcodeproj/project.pbxproj
--- A/Libsystem-1252/Libsystem.xcodeproj/project.pbxproj	2016-10-18 00:11:33.000000000 +0100
+++ B/Libsystem-1252/Libsystem.xcodeproj/project.pbxproj	2018-07-02 12:14:38.000000000 +0100
@@ -276,17 +276,19 @@
 /* Begin XCBuildConfiguration section */
 		BA5511AF136C9D400012B9B5 /* Release */ = {
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = C9B56C55153751A600C153A8 /* Libsystem.xcconfig */;
 			buildSettings = {
+				SDKROOT = macosx.internal;
 			};
 			name = Release;
 		};
 		BA5511C6136C9E2E0012B9B5 /* Release */ = {
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = C9B56C55153751A600C153A8 /* Libsystem.xcconfig */;
 			buildSettings = {
+				GCC_PREPROCESSOR_DEFINITIONS = __PUREDARWIN__;
 				PRODUCT_NAME = "$(TARGET_NAME).B";
 				"PRODUCT_NAME[sdk=iphonesimulator*]" = "$(TARGET_NAME)";
 				SDKROOT = macosx.internal;
 			};
 			name = Release;
Only in B/Libsystem-1252/Libsystem.xcodeproj/project.xcworkspace: xcshareddata
Only in B/Libsystem-1252/Libsystem.xcodeproj/project.xcworkspace: xcuserdata
Only in B/Libsystem-1252/Libsystem.xcodeproj: xcuserdata
diff -rwU5 A/Libsystem-1252/init.c B/Libsystem-1252/init.c
--- A/Libsystem-1252/init.c	2017-09-26 17:29:44.000000000 +0100
+++ B/Libsystem-1252/init.c	2018-07-02 12:14:38.000000000 +0100
@@ -45,15 +45,17 @@
 extern void __pthread_init(const struct _libpthread_functions *libpthread_funcs, const char *envp[], const char *apple[], const struct ProgramVars *vars);	// from libsystem_pthread.dylib
 extern void __malloc_init(const char *apple[]); // from libsystem_malloc.dylib
 extern void __keymgr_initializer(void);		// from libkeymgr.dylib
 extern void _dyld_initializer(void);		// from libdyld.dylib
 extern void libdispatch_init(void);		// from libdispatch.dylib
+#if !defined(__PUREDARWIN__)
 extern void _libxpc_initializer(void);		// from libxpc.dylib
 extern void _libsecinit_initializer(void);        // from libsecinit.dylib
 extern void _libtrace_init(void);		// from libsystem_trace.dylib
 extern void _container_init(const char *apple[]); // from libsystem_containermanager.dylib
 extern void __libdarwin_init(void);		// from libsystem_darwin.dylib
+#endif /* !defined(__PUREDARWIN__ */
 
 
 // signal malloc stack logging that initialisation has finished
 extern void __stack_logging_early_finished(void); // form libsystem_c.dylib
 
@@ -71,31 +73,37 @@
 
 extern void dispatch_atfork_prepare(void);
 extern void dispatch_atfork_parent(void);
 extern void dispatch_atfork_child(void);
 
+#if !defined(__PUREDARWIN__)
 extern void _libtrace_fork_child(void);
+#endif /* !defined(__PUREDARWIN__) */
 
 extern void _malloc_fork_prepare(void);
 extern void _malloc_fork_parent(void);
 extern void _malloc_fork_child(void);
 
 extern void _mach_fork_child(void);
 extern void _notify_fork_child(void);
 extern void _dyld_fork_child(void);
+#if !defined(__PUREDARWIN__)
 extern void xpc_atfork_prepare(void);
 extern void xpc_atfork_parent(void);
 extern void xpc_atfork_child(void);
 extern void _libSC_info_fork_prepare(void);
 extern void _libSC_info_fork_parent(void);
 extern void _libSC_info_fork_child(void);
+#endif /* !defined(__PUREDARWIN__) */
 extern void _asl_fork_child(void);
 
 #if defined(HAVE_SYSTEM_CORESERVICES)
 // libsystem_coreservices.dylib
 extern void _libcoreservices_fork_child(void);
+#if !defined(__PUREDARWIN__)
 extern char *_dirhelper(int, char *, size_t);
+#endif /* !defined(__PUREDARWIN__) */
 #endif
 
 #if TARGET_OS_EMBEDDED && !TARGET_OS_WATCH && !__LP64__
 extern void _vminterpose_init(void);
 #endif
@@ -138,13 +146,13 @@
 	static const struct _libc_functions libc_funcs = {
 		.version = 1,
 		.atfork_prepare = libSystem_atfork_prepare,
 		.atfork_parent = libSystem_atfork_parent,
 		.atfork_child = libSystem_atfork_child,
-#if defined(HAVE_SYSTEM_CORESERVICES)
+#if defined(HAVE_SYSTEM_CORESERVICES) && !defined(__PUREDARWIN__)
 		.dirhelper = _dirhelper,
-#endif
+#endif /* defined(HAVE)SYSTEM_CORESERVICES) && !defined(__PUREDARWIN__) */
 	};
 
 	__libkernel_init(&libkernel_funcs, envp, apple, vars);
 
 	__libplatform_init(NULL, envp, apple, vars);
@@ -162,24 +170,29 @@
 #endif
 
 	_dyld_initializer();
 
 	libdispatch_init();
+	
+#if !defined(__PUREDARWIN__)
 	_libxpc_initializer();
 
 	// must be initialized after dispatch
 	_libtrace_init();
 
 #if !(TARGET_OS_EMBEDDED || TARGET_OS_SIMULATOR)
 	_libsecinit_initializer();
 #endif
+#endif /* !defined(__PUREDARWIN__) */
 
 #if TARGET_OS_EMBEDDED
 	_container_init(apple);
 #endif
 
+#if !defined(__PUREDARWIN__)
 	__libdarwin_init();
+#endif /* !defined(__PUREDARWIN__) */
 
 	__stack_logging_early_finished();
 
 #if TARGET_OS_EMBEDDED && TARGET_OS_IOS && !__LP64__
 	_vminterpose_init();
@@ -214,12 +227,14 @@
 	// first call client prepare handlers registered with pthread_atfork()
 	_pthread_atfork_prepare_handlers();
 
 	// second call hardwired fork prepare handlers for Libsystem components
 	// in the _reverse_ order of library initalization above
+	#if !defined(__PUREDARWIN__)
 	_libSC_info_fork_prepare();
 	xpc_atfork_prepare();
+	#endif /* !defined(__PUREDARWIN__) */
 	dispatch_atfork_prepare();
 	_malloc_fork_prepare();
 	_pthread_atfork_prepare();
 }
 
@@ -229,12 +244,14 @@
 	// first call hardwired fork parent handlers for Libsystem components
 	// in the order of library initalization above
 	_pthread_atfork_parent();
 	_malloc_fork_parent();
 	dispatch_atfork_parent();
+	#if !defined(__PUREDARWIN__)
 	xpc_atfork_parent();
 	_libSC_info_fork_parent();
+	#endif /* !defined(__PUREDARWIN__) */
 
 	// second call client parent handlers registered with pthread_atfork()
 	_pthread_atfork_parent_handlers();
 }
 
@@ -247,18 +264,20 @@
 	_pthread_atfork_child();
 	_mach_fork_child();
 	_malloc_fork_child();
 	_libc_fork_child(); // _arc4_fork_child calls malloc
 	dispatch_atfork_child();
-#if defined(HAVE_SYSTEM_CORESERVICES)
+#if defined(HAVE_SYSTEM_CORESERVICES) && !defined(__PUREDARWIN__)
 	_libcoreservices_fork_child();
 #endif
 	_asl_fork_child();
 	_notify_fork_child();
+	#if !defined(__PUREDARWIN__)
 	xpc_atfork_child();
 	_libtrace_fork_child();
 	_libSC_info_fork_child();
+	#endif /* !defined(__PUREDARWIN__) */
 
 	// second call client parent handlers registered with pthread_atfork()
 	_pthread_atfork_child_handlers();
 }
 
Only in B/Libsystem-1252: obj
diff -rwU5 A/Libsystem-1252/requiredlibs B/Libsystem-1252/requiredlibs
--- A/Libsystem-1252/requiredlibs	2017-04-28 09:39:09.000000000 +0100
+++ B/Libsystem-1252/requiredlibs	2018-07-02 12:14:37.000000000 +0100
@@ -1,29 +1,21 @@
-cache
 commonCrypto
 compiler_rt
 copyfile
-corecrypto
 dispatch
 dyld
 launch
 macho
 removefile
 system_asl
 system_blocks
 system_c
-system_configuration
 system_coreservices
-system_darwin
 system_dnssd
 system_info
 system_sim_kernel system_kernel
 system_m
 system_malloc
-system_network
 system_notify
 system_sim_platform system_platform
 system_sim_pthread system_pthread
-system_sandbox
-system_trace
 unwind
-xpc
Only in B/Libsystem-1252: sym
